tests/e2e/flowtest/test_base_operation_node_restart_kms.py

```diff
@@ -0,0 +1,210 @@
+import pytest
+import logging
+from time import sleep
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_bm,
+    skipif_aws_i3,
+    skipif_no_kms,
+)
+from ocs_ci.ocs import node, constants, defaults
+from ocs_ci.framework.testlib import config, E2ETest, flowtests, ignore_leftovers
+from ocs_ci.ocs import flowtest
+from ocs_ci.helpers.disruption_helpers import Disruptions
+from ocs_ci.ocs.bucket_utils import BucketPolicyOps, ObcIOs
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.ocs.resources import pod
+
+logger = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@flowtests
+class TestBaseOperationNodeRestart(E2ETest):
+    """
+    Tests Story/Flow based test scenario: Node Restart
+
+    """
+
+    @skipif_aws_i3
+    @skipif_bm
+    @skipif_no_kms
+    @pytest.mark.polarion_id("OCS-2487")
+    def test_base_operation_node_restart(
+        self,
+        node_drain_teardown,
+        node_restart_teardown,
+        nodes,
+        project_factory,
+        pgsql_factory_fixture,
+        couchbase_factory_fixture,
+        multi_pvc_factory,
+        mcg_obj,
+        bucket_factory,
+    ):
+        """
+        Test covers following flow operations while running workloads in the background:
+        1. Node reboot
+        2. Device replacement
+        3. Bucket policy: put, modify, delete
+        4. Node drain
+        5. Node n/w failure
+
+        """
+        logger.info("nooba kms validation..")
+        noobaa_kms_validation()
```

Comment:
IIUC, we are restricting the test to run on only KMS here. We can make this script to run even on non kms cluster by adding some if_kms checks in the test, that way we can make it more generic so that it can run on both kms and non-kms . wdyt?

Reply:
got it